### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by mishagitcode**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,3 +1,91 @@
...
+
+    def __str__(self):
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self):
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance | int | float):
+
+        if other is not None:
```
Comment (ArsenPidhoretskyi) [inline app/main.py:16]: It is always true

File: app/main.py [hunk 2]
```diff
@@ -1,3 +1,91 @@
...
+
+            if isinstance(other, Distance):
+                return Distance(self.km + other.km)
+
+            if isinstance(other, int) or isinstance(other, float):
```
Comment (ArsenPidhoretskyi) [inline app/main.py:21]:
```suggestion
            if isinstance(other, (int, float):
```

File: app/main.py (hunk 1 above)
Comment (ArsenPidhoretskyi) [reply app/main.py:16]: Fix in all places

File: app/main.py (hunk 2 above)
Comment (ArsenPidhoretskyi) [reply app/main.py:21]: Fix in all places

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]
Provide return type annotations

File: app/main.py [hunk 3]
```diff
@@ -1,3 +1,80 @@
...
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance | int | float) -> Distance:
+
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+
+        if isinstance(other, (int, float)):
```
Comment (EdAlekseiev) [inline app/main.py:19]: This if-statement is redundant

File: app/main.py [hunk 4]
```diff
@@ -1,3 +1,80 @@
...
+            return Distance(self.km + other)
```
Comment (EdAlekseiev) [inline app/main.py:20]:
```suggestion
        return Distance(self.km + other)
```

File: app/main.py [hunk 5]
```diff
@@ -1,3 +1,80 @@
...
+
+    def __iadd__(self, other: Distance | int | float) -> Distance:
+
+        if isinstance(other, Distance):
+            self.km += other.km
+            return self
+
+        if isinstance(other, (int, float)):
+            self.km += other
+            return self
```
Comment (EdAlekseiev) [inline app/main.py:30]: This if-statement is redundant

Comment (EdAlekseiev) [inline app/main.py:30]:
```suggestion
        self.km += other
        return self
```

File: app/main.py [hunk 6]
```diff
@@ -1,3 +1,80 @@
...
+        if isinstance(other, (int, float)):
+            return Distance(round(self.km / other, 2))
+
+    def __eq__(self, other: Distance | int | float) -> bool:
+
+        if isinstance(other, Distance):
+            return self.km == other.km
+
+        if isinstance(other, (int, float)):
+            return self.km == other
```
Comment (EdAlekseiev) [inline app/main.py:48]: This if-statement is redundant

Comment (EdAlekseiev) [inline app/main.py:48]:
```suggestion
        return self.km == other
```

File: app/main.py [hunk 7]
```diff
@@ -1,3 +1,80 @@
...
+        if isinstance(other, (int, float)):
+            return self.km >= other
+
+    def __le__(self, other: Distance | int | float) -> bool:
+
+        if isinstance(other, Distance):
+            return self.km <= other.km
+
+        if isinstance(other, (int, float)):
+            return self.km <= other
```
Comment (EdAlekseiev) [inline app/main.py:80]: Please, delete the redundant if-statements, as I showed above

**Review by EdAlekseiev — CHANGES_REQUESTED** [review]

**Review by YukihiroSM — APPROVED** [review]
Good job!

**Pull request closed without merging**